### PR TITLE
fix history link

### PR DESCRIPTION
**Pull request by eroshenkoam**

### CODE REVIEW

File: allure-generator/src/main/java/io/qameta/allure/history/HistoryPlugin.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ private void updateHistory(final Map<String, HistoryData> history,
                 .withTime(result.getTime());
 
         if (Objects.nonNull(info.getReportUrl())) {
-            newItem.setReportUrl(String.format("%s/#testcase/%s", info.getReportUrl(), result.getUid()));
+            newItem.setReportUrl(String.format("%s#testcase/%s", info.getReportUrl(), result.getUid()));
```
Comment (baev) [inline allure-generator/src/main/java/io/qameta/allure/history/HistoryPlugin.java]: Better to use conditional link behaviour instead:

```java
if (info.ReportUrl().endsWith('/index.html') || info.ReportUrl().endsWith('/')) { //and handle NPE as well
   ...
}
```

File: allure-generator/src/main/java/io/qameta/allure/history/HistoryPlugin.java [hunk 2]
```diff
@@ -37,7 +37,8 @@
 
     //@formatter:off
     private static final TypeReference<Map<String, HistoryData>> HISTORY_TYPE =
-        new TypeReference<Map<String, HistoryData>>() {};
+            new TypeReference<Map<String, HistoryData>>() {
```
Comment (baev) [inline allure-generator/src/main/java/io/qameta/allure/history/HistoryPlugin.java]: remove this change

PS you can configure IDEA: go to Editor -> Code Style -> Enable formatter markers in comments

File: allure-generator/src/main/java/io/qameta/allure/history/HistoryPlugin.java [hunk 3]
```diff
@@ -89,8 +90,8 @@ private void updateHistory(final Map<String, HistoryData> history,
                                final ExecutorInfo info) {
         //@formatter:off
         final HistoryData data = history.computeIfAbsent(
-            result.getHistoryId(),
-            id -> new HistoryData().withStatistic(new Statistic())
+                result.getHistoryId(),
```
Comment (baev) [inline allure-generator/src/main/java/io/qameta/allure/history/HistoryPlugin.java]: remove this change

File: allure-generator/src/main/java/io/qameta/allure/history/HistoryPlugin.java (hunk 2 above)
Comment (eroshenkoam) [inline allure-generator/src/main/java/io/qameta/allure/history/HistoryPlugin.java]: ty, I didn't know it

**Pull request merged**